framework/core/module_test.ts

```diff
@@ -0,0 +1,52 @@
+
+import { assertEquals, assertThrows } from 'https://deno.land/std@0.90.0/testing/asserts.ts'
+import { trimModuleExt, toPagePath, importModule } from './module.ts'
+
+// Augment the window object
+interface MockWindow extends Window {
+    __ALEPH: { import: (src: string, refetch?: boolean) => string | undefined }
+}
+declare let window: MockWindow
+
+Deno.test('module toPagePath', () => {
+    assertEquals(toPagePath('/pages/bar.ts'), '/bar')
+    assertEquals(toPagePath('/foobar/index.ts'), '/foobar')
+
+    assertEquals(toPagePath(''), '/')
+
+    assertThrows(() => toPagePath({} as string))
+})
+
+Deno.test('module trimModuleExt', () => {
+    assertEquals(trimModuleExt('foobar.ts'), 'foobar')
+
+    assertEquals(trimModuleExt('baz.zip'), 'baz.zip')
+
+    assertEquals(trimModuleExt('barbaz.jsx.ts'), 'barbaz.jsx')
+
+    assertEquals(trimModuleExt('/index.ts'), '/index')
+
+    assertThrows(() => trimModuleExt({} as string))
+
+    assertEquals(trimModuleExt('😀'), '😀')
+
+})
+
+Deno.test('module importModule', async () => {
+    // mock  __ALEPH.import
+    function foo(str: string, refetch: boolean = false) {
+        if (refetch) {
+            return 'refetching...'
+        }
+        return str
```

Comment:
Don't we need to return a promise?

Reply:
It was not necessary since __ALEPH.import does not currently have a type definition, but I made that change in commit 75664fc3d5c just in case that happens in the future.

Reply:
@cdoremus Actually, type definitions is not the reason. The goal of mocks like this is to mimic the API 100% and so if we didn't return a promise that probably won't change much, but it is still good to have.